src/main/java/org/cascadebot/cascadebot/moderation/ModlogEvent.java

```diff
@@ -0,0 +1,184 @@
+    CASCADE_PERMISSIONS_GROUP_PERMISSION_REMOVE(MessageType.WARNING, Category.CASCADE, Category.CASCADE_PERMISSIONS),
+    CASCADE_PERMISSIONS_GROUP_LINK(MessageType.WARNING, Category.CASCADE, Category.CASCADE_PERMISSIONS),
+
+    CASCADE_PERMISSIONS_USER_PERMISSION_ADD(MessageType.WARNING, Category.CASCADE, Category.CASCADE_PERMISSIONS),
+    CASCADE_PERMISSIONS_USER_PERMISSION_REMOVE(MessageType.WARNING, Category.CASCADE, Category.CASCADE_PERMISSIONS),
+    CASCADE_PERMISSIONS_USER_GROUP_ADD(MessageType.WARNING, Category.CASCADE, Category.CASCADE_PERMISSIONS),
+    CASCADE_PERMISSIONS_USER_GROUP_REMOVE(MessageType.WARNING, Category.CASCADE, Category.CASCADE_PERMISSIONS),
+
+    CASCADE_SETTINGS_UPDATED(MessageType.WARNING, Category.CASCADE),
+
+    CASCADE_MODULE_UPDATED(MessageType.WARNING, Category.CASCADE),
+
+    CASCADE_COMMAND_RUN(MessageType.INFO, Category.CASCADE),
+    CASCADE_COMMAND_RUN_ERROR(MessageType.WARNING, Category.CASCADE),
+
+    CASCADE_PLAYLIST_CREATED(MessageType.INFO, Category.CASCADE),
+    CASCADE_PLAYLIST_DELETED(MessageType.INFO, Category.CASCADE),
+
+    CASCADE_TEMP_MUTE(MessageType.DANGER, Category.CASCADE, Category.CASCADE_MODERATION, Category.MODERATION),
+    CASCADE_TEMP_BAN(MessageType.DANGER, Category.CASCADE, Category.CASCADE_MODERATION, Category.MODERATION),
+    CASCADE_SOFT_BAN(MessageType.DANGER, Category.CASCADE, Category.CASCADE_MODERATION, Category.MODERATION),
+    CASCADE_PURGE(MessageType.DANGER, Category.CASCADE, Category.CASCADE_MODERATION, Category.MODERATION),
+
+    CASCADE_WHITELIST(MessageType.WARNING, Category.CASCADE, Category.CASCADE_MODERATION, Category.MODERATION),
+    CASCADE_BLACKLIST(MessageType.WARNING, Category.CASCADE, Category.CASCADE_MODERATION, Category.MODERATION),
+
+    CASCADE_SCRIPT_CREATED(MessageType.WARNING, Category.CASCADE, Category.CASCADE_CUSTOM_COMMANDS),
+    CASCADE_SCRIPT_DELETED(MessageType.WARNING, Category.CASCADE, Category.CASCADE_CUSTOM_COMMANDS),
+    CASCADE_SCRIPT_UPDATED(MessageType.WARNING, Category.CASCADE, Category.CASCADE_CUSTOM_COMMANDS),
+
+    CASCADE_TAG_CREATED(MessageType.WARNING, Category.CASCADE, Category.CASCADE_CUSTOM_COMMANDS),
+    CASCADE_TAG_DELETED(MessageType.WARNING, Category.CASCADE, Category.CASCADE_CUSTOM_COMMANDS),
+    CASCADE_TAG_UPDATED(MessageType.WARNING, Category.CASCADE, Category.CASCADE_CUSTOM_COMMANDS);
+
+    @Getter
+    private static Map<Category, List<ModlogEvent>> modlogCategoryMap = new HashMap<>();
+
+    static {
+        modlogCategoryMap.put(Category.ALL, Arrays.asList(ModlogEvent.values()));
+        for (ModlogEvent event : ModlogEvent.values()) {
+            for (Category category : event.getCategories()) {
+                if (modlogCategoryMap.containsKey(category)) {
+                    modlogCategoryMap.get(category).add(event);
+                } else {
+                    modlogCategoryMap.put(category, new ArrayList<>(Set.of(event)));
+                }
+            }
+        }
+    }
```

Comment:
This would be so much simpler in Kotlin

```kotlin
modlogCategoryMap[Category.ALL] = ModlogEvent.values().toList()
modlogCategoryMap.putAll(ModlogEvent.values().groupBy { it.category })
```

Reply:
You made this originally lol.